### PR TITLE
twister: fix setting status on unprocessed tests

### LINKED ISSUE #45845: tests: The failure test case number increase significantly in CMSIS DSP tests on ARM boards.
**Describe the bug**
After the twister overhaul PR https://github.com/zephyrproject-rtos/zephyr/pull/44094, it reveal some incorrect testcase numbers in the twister report. 
Take frdm_k64f as an example, following tests are affected most:  
tests/lib/cmsis_dsp/martix  => 5940   failure
tests/lib/cmsis_dsp/transform  => 1100  failure
tests/lib/cmsis_dsp/filtering => 2172 failure

Because that testcases cannot be parsed well when twister to scan the test case from source code. We did not see this issue because
these incorrect testcases was judge as "skipped" instead of "failure" before this PR, but issues existed before.

**To Reproduce**
1. source zephry-env.sh
2. twister -p frdm_k64f --device-testing --device-serial=/dev/ttyACM0  -T tests/lib/cmsis_dsp/ -v

**Expected behavior**
The failure testcases should be very less.

**Impact**
The failure testcases increase significantly in the test report.

**Logs and console output**
```
<testsuites><testsuite name="frdm_k64f" timestamp="2022-05-18T01:48:09.248298" time="115.16999999999999" tests="15000" failures="9282" errors="0" skipped="154"><properties>
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain (e.g Zephyr SDK 0.14.2)
 - Commit SHA: d78194546746

### PR DESCRIPTION
**Pull request by nashif**
- twister: do not mark tests as blocked if a testsuite has passed
- twister: do not set case status on flash timeout

Fixes #45845

### CODE REVIEW

**Review by stephanosio — DISMISSED** [review]

File: scripts/pylib/twister/twisterlib.py [hunk 1]
```diff
@@ -961,11 +961,6 @@ def handle(self):
         if harness.is_pytest:
             harness.pytest_run(self.log)
 
-        # sometimes a test instance hasn't been executed successfully with no
```
Comment (LixinGuoX) [inline scripts/pylib/twister/twisterlib.py:964]: I am wondering where else can we process those testcases with no status if the suite get passed, or just leave them as status null?

Comment (enjiamai) [reply scripts/pylib/twister/twisterlib.py:964]: Hi, @nashif , may I confirm this with you? Were you on purpose to make these testcases' status as "null", then trying to fix them in test case code later? Thanks!

Comment (nashif) [reply scripts/pylib/twister/twisterlib.py:964]: those cases will later be processed in `xunit_testcase` and get the right explanation which is a hint to fix the testsuite and make it compatible with twister.

**Review by enjiamai — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: scripts/pylib/twister/twisterlib.py (hunk 1 above)
Comment (LixinGuoX) [reply scripts/pylib/twister/twisterlib.py:964]: But we don't have similar process in `json_report`, they are still marked as `null` in twister.json, should we add this to `json_report`?

Comment (LixinGuoX) [reply scripts/pylib/twister/twisterlib.py:964]: Ok, thanks!

File: scripts/pylib/twister/twisterlib.py [hunk 2]
```diff
@@ -1707,13 +1703,15 @@ def __eq__(self, other):
...
 class TestCase(DisablePyTestCollectionMixin):
+
     def __init__(self, name=None, testsuite=None):
         self.duration = 0
         self.name = name
         self.status = None
         self.reason = None
         self.testsuite = testsuite
         self.output = ""
+        self.freeform = False
```
Comment (gopiotr) [inline scripts/pylib/twister/twisterlib.py:1714]: `freeform` attribute is always set just after create `TestCase` object, so maybe it should be passed as an argument to `__init__` function, like this:
```python
def __init__(self, name=None, testsuite=None, freeform=False):
        self.duration = 0
        self.name = name
        self.status = None
        self.reason = None
        self.testsuite = testsuite
        self.output = ""
        self.freeform = freeform
```

File: scripts/pylib/twister/twisterlib.py [hunk 3]
```diff
@@ -2152,8 +2151,9 @@ def set_case_status_by_name(self, name, status, reason=None):
             tc.reason = reason
         return tc
 
-    def add_testcase(self, name):
+    def add_testcase(self, name, freeform=False):
         tc = TestCase(name=name)
+        tc.freeform = freeform
```
Comment (gopiotr) [inline scripts/pylib/twister/twisterlib.py:2156]: As I mentioned earlier, `freeform` argument could be passed with `__init__` function:
```python
tc = TestCase(name=name, testsuite=self, freeform=freeform)
```

**Review by chen-png — APPROVED** [review]

**Pull request merged**